File: ios/Kwift Template/com/lightningkite/kwifttemplate/shared/api/APIOnline.swift
//Package: com.lightningkite.kwifttemplate.shared.api
//Converted using Kwift2

import Foundation
import Kwift
import KwiftMaps



public class APIOnline: APIInterface {
    
    public var baseUrl: String
    
    
    public func getExamplePosts(onResult: @escaping (Int32, Array<Post>?, String?) -> Void) -> Void {
        HttpClient.call(url: baseUrl + "/posts", method: HttpClient.GET, headers: [:], body: nil, onResult: onResult)
    }
    
    public init(baseUrl: String = "https://jsonplaceholder.typicode.com") {
        self.baseUrl = baseUrl
    }
    convenience public init(_ baseUrl: String) {
        self.init(baseUrl: baseUrl)
    }
}
 

File: android/src/main/java/com/lightningkite/butterflytemplate/api/APIOnline.kt
//! This file will translate using Khrysalis.
///Kotlin Only
@file:Suppress("ConvertToStringTemplate", "RemoveExplicitTypeArguments")

///End Kotlin Only

package com.lightningkite.butterflytemplate.api

import com.lightningkite.butterfly.net.HttpClient
import com.lightningkite.butterfly.net.readJson
import com.lightningkite.butterflytemplate.models.Post
import io.reactivex.Single


class APIOnline(val baseUrl: String = "https://jsonplaceholder.typicode.com") : APIInterface {
    override fun getExamplePosts(): Single<List<Post>> {
        return HttpClient.call(
            url = baseUrl + "/posts",
            method = HttpClient.GET,
            headers = mapOf()
        ).readJson()
    }

}
